File: 0217-contains-duplicate/0217-contains-duplicate.py
class Solution:
    def containsDuplicate(self, nums: List[int]) -> bool:
        hash = set()
        for num in nums: # O(1) Time complexity for search
            if num in hash:
                return True
            else:
                hash.add(num)
        return False
        # Overall TC = O(n)
        # Overall SC = O(n) -> for storing num in a set

File: 0217-contains-duplicate/0217-contains-duplicate.java
class Solution {
    public boolean containsDuplicate(int[] nums) {
        HashSet<Integer> hash = new HashSet<>();
	    for(int num : nums) {
	         if(!hash.add(num))  // add method returns false if num is already in the set
	            return true;  
	    }
		return false;
    }
}